File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        // 本質上找出最小和最大值即可
        // 但是因為時間的順序關係，因此
        int left = prices[0];
        int res = 0;
        for (int i = 0; i < prices.length; ++i) {
            if (prices[i] < left) { left = prices[i]; }
            if (prices[i] - left > res) { res = prices[i] - left; }
        }
        return res;
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums) - 1
        mid = (left + right) // 2
        res = -1 

        while left <= right:
            if nums[mid] == target:
                res = mid
                break
            if nums[mid] < target:
                left = mid + 1
                mid = (left + right) // 2
            elif nums[mid] > target:
                right = mid - 1
                mid = (left + right) // 2 
        return res 
        